### ISSUE TITLE
Notion code blocks are imported without line breaks

### ISSUE THREAD

**Issue by smpb** [obsidianmd/obsidian-importer#82]
Greetings.
When using the importer on a Notion backup, code blocks that span multiple lines are erroneously recreated as a single line of text.

For example, this is a page with a couple of code blocks in Notion:
<img width="720" alt="Notion Screenshot" src="https://github.com/obsidianmd/obsidian-importer/assets/168636/062df8ab-7d0d-45c1-9e7d-6ee81a8ac659">

And this is the result of importing that same page to Obsidian:
<img width="720" alt="Obsidian Screenshot" src="https://github.com/obsidianmd/obsidian-importer/assets/168636/a73411aa-d4b7-405a-9ce4-7c1bf15f4a0d">

I believe this to be an incorrect behaviour since it invalidates the formatting and indentation of every single code block in a vault, and would greatly appreciate a way to have it behave the expected way.

**Comment by wysstobi** [comment]
I reviewed the code two days ago since I ran into the same issue. 
The importer plugin replaces all newlines (\n) with <br />. This seems not to be a problem at first sight, but somehow <br /> tags inside of <code> tags are ignored by the function `htmltoMarkdown` which uses the Turndown Service. 

if you clone this repository and remove the function call on line 39 (`encodeNewlinesToBr(body);`) in the file `convert-to-md.ts` you will see this is not broken anymore. 

It might even be a work around for you, since I did not see a difference in the result by just removing this function call. 

To completely fix it, a more sophisticated version of this function could be implemented which doesn not replace new lines in code blocks. But I was not able to figure out a regular expression which works in all cases (especially when code blocks are nested in other elements as for examples in lists).

**Comment by smpb** [comment]
> To completely fix it, a more sophisticated version of this function could be implemented which doesn not replace new lines in code blocks. But I was not able to figure out a regular expression which works in all cases (especially when code blocks are nested in other elements as for examples in lists).

Thanks for doing the groundwork and pointing me in the right direction.
I changed the `encodeNewLinestoBr` function to the following:

```diff
diff --git a/src/formats/notion/convert-to-md.ts b/src/formats/notion/convert-to-md.ts
index 3e7163e..4af7609 100644
--- a/src/formats/notion/convert-to-md.ts
+++ b/src/formats/notion/convert-to-md.ts
@@ -267,7 +267,7 @@ function replaceTableOfContents(body: HTMLElement) {
 }

 function encodeNewlinesToBr(body: HTMLElement) {
-       body.innerHTML = body.innerHTML.replace(/\n/g, '<br />');
+       body.innerHTML = body.innerHTML.replace(/\n(?=((?!<\/code).)*?(<code|$))/gs, '<br />');
 }

 function stripLinkFormatting(body: HTMLElement) {
```

I've had positive results with this change (even with some more elaborate edge-cases), but be aware that I've not tested it fully.  Could be useful, though, so I'm adding it here.

Thanks again.

**Comment by joshuatazrein** [comment]
Thanks for the RegEx. I added a reversal of code block formatting:

```
body.innerHTML = body.innerHTML.replace(/\n/g, '<br />');
	// Since <br /> is ignored in codeblocks, we replace with newlines
	const codeBlocks = body.findAll('code')
	codeBlocks.forEach(block => block.innerHTML = block.innerHTML.replace(/<br \\\/>/g, '\n'))
```

**Comment by aksh1618** [comment]
Hey there! Notion import has been the main roadblock for me to switch to Obsidian, so thanks for adding this functionality! I tried to import my workspace today, however I'm still facing this issue. I confirmed that the version of the plugin being used is 1.3.1, which should have this fix, however all my code blocks are still ending up without any line breaks (And my notion is absolutely full of them). Let me know if I can provide anything to help debug this further!

![image](https://github.com/obsidianmd/obsidian-importer/assets/26065812/c207dfd5-2220-412d-8422-f8a211f107d0)
